pages/index.tsx

```diff
@@ -1,72 +1,184 @@
-import type { NextPage } from 'next'
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import {
+  Card,
+  CardContent,
+  TextField,
+  Button,
+  Container,
+  Box,
+  CardMedia,
+  Typography,
+  CircularProgress,
+  ThemeProvider,
+  createTheme,
+} from '@mui/material';
 
-const Home: NextPage = () => {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
-
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+import type { NextPage } from 'next';
+import { ChangeEvent, useMemo, useState } from 'react';
+import { GetEmailAddress } from '../interfaces/email';
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.tsx</code>
-        </p>
+const theme = createTheme({
+  palette: {
+    primary: {
+      main: '#00ffa3',
+    },
+  },
+});
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
-
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h2>Learn &rarr;</h2>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+const Home: NextPage = () => {
+  const [email, setEmail] = useState<string | null>(null);
+  const [error, setError] = useState(false);
```

Comment:
```suggestion
  const [error, _setError] = useState(false);
```
If not needed just remove it

Reply:
```suggestion
  const [hasError, setHasError] = useState<boolean>(false);
```